compression.go

```diff
@@ -0,0 +1,71 @@
+package kafka
+
+const (
+	CompressionNone int8 = iota
+	CompressionGZIP
+	CompressionSnappy
+	CompressionLZ4
+)
+
+var codecs map[int8]CompressionCodec
+
+// RegisterCompressionCodec registers a compression codec so it can be used by a Writer.
+func RegisterCompressionCodec(codec func() CompressionCodec) {
+	if codecs == nil {
```

Comment:
What's the point of this lazy-allocation here? Why not simply create the map when it's declared?
```go
var codecs = make(map[int8]CompressionCodec)
```